stackrox/stackrox
# ROX-9588: Limit email lines length to RFC 5322 specification

## Description

We have two important content of emails that should be limited.
1. base64 encoded attachments
2. text of the email

The Following things are added:
1. for attachments, we are chunking content on 78 chars length and adding `\r\n` to them.
2. for text in the email - we are doing word wrapping to a line length of 78 characters

## Checklist
- ~[ ] Evaluated and added CHANGELOG entry if required~ - probably not relevant information for the users
- ~[ ] Determined and documented upgrade steps~ - no required upgrade steps
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~ - not user facing changes

## Testing Performed

**Test Vulnerability reports**
1. add MailHog to the cluster. Guide: https://fabianlee.org/2022/01/16/kubernetes-running-a-mail-container-for-testing-email-during-development/
3. create an email notifier to use MailHog. Important information from guide are: `node_ip` and `smtp_port`. **Note:** in the guide `@me` is used, but we can use `@me.com` for testing because our fields are checking if it's an email address with a valid domain.
4. create reporting that will use created email notifier
5. trigger "run report" from UI
6. check in MailHog UI the email. Open the "Source" tab and validate that line lengths are 78 chars.

**Test Policy**
1. search for "Latest tag" policy - URI: `/main/policies?s[Policy]=latest` and edit it
2. add a previously created email notifier -> and save the policy
3. create a deployment with the "latest" tag - i.e. `k create deployment nginx-latest --image=nginx:latest`
4. after a few seconds -> there should be an e-mail in MailHog UI and on the "Source" tab we can validate that line lengths are 78 chars.

**Testing of GMail Web UI - and format of email**
1. create an email notifier with the Google SMTP server and use your Google email as the default email address
2. and follow the previously explained "Test Policy" testing until step 4.
3. open your GMail and check the received email
